graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/aggregations/AggregationsFetcherSupport.java

```diff
@@ -0,0 +1,210 @@
+            putResultValue(columns, row, selectedField, Row::getLong);
+            break;
+          case DECIMAL_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getBigDecimal);
+            break;
+          case VARINT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getBigInteger);
+            break;
+          case FLOAT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getFloat);
+            break;
+          case SMALLINT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getShort);
+            break;
+          case TINYINT_FUNCTION:
+            putResultValue(columns, row, selectedField, Row::getByte);
+            break;
+        }
+      }
+    }
+    return columns;
+  }
+
+  private void putResultValue(
+      Map<String, Object> columns,
+      Row row,
+      SelectedField selectedField,
+      BiFunction<Row, String, Object> rowValueExtractor) {
+    String alias = selectedField.getAlias();
+    // put the returned value as alias
+    if (alias != null) {
+      columns.put(selectedField.getName(), rowValueExtractor.apply(row, alias));
+    }
+    // generate aggregation column name and get the value
+    else {
+      String columnName = generateAggregationColumnName(selectedField);
+      columns.put(selectedField.getName(), rowValueExtractor.apply(row, columnName));
+    }
+  }
+
+  // it returns C* generated aggregation column name
+  // for example, for the name = count and arguments = ['a']
+  // it will return: 'system.count(a)'
+  private String generateAggregationColumnName(SelectedField selectedField) {
+    Map<String, Object> arguments = selectedField.getArguments();
+    SupportedAggregationFunction functionName = getNameArgument(arguments);
+    List<String> args = getAndValidateArgs(arguments, functionName);
+    return String.format("system.%s(%s)", functionName.getName(), args.get(0));
+  }
+
+  private void addAggregation(
+      Map<String, Object> arguments,
+      QueryBuilder.QueryBuilder__20 queryBuilder,
+      BiFunction<QueryBuilder.QueryBuilder__20, String, QueryBuilder.QueryBuilder__19>
+          addAggregation,
+      SelectedField selectedField,
+      SupportedAggregationFunction supportedAggregationFunction) {
+    List<String> args = getAndValidateArgs(arguments, supportedAggregationFunction);
+    String column = getAndValidateColumn(args);
+
+    String alias = selectedField.getAlias();
+    QueryBuilder.QueryBuilder__19 withAggregation = addAggregation.apply(queryBuilder, column);
+    if (alias != null) {
+      withAggregation.as(alias);
+    }
+  }
+
+  private String getAndValidateColumn(List<String> args) {
+    String column = dbColumnGetter.getDBColumnName(table, args.get(0));
+    if (column == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The column name: %s provided for the %s function does not exists.",
+              args.get(0), COUNT));
```

Comment:
I don't think you want `COUNT` here, right?

Reply:
good catch, fixed